olp-cpp-sdk-core/src/client/PendingRequests.cpp

```diff
@@ -33,17 +32,33 @@ PendingRequests::PendingRequests(){};
 
 PendingRequests::~PendingRequests(){};
 
-bool PendingRequests::CancelPendingRequests() {
+bool PendingRequests::CancelAll() {
   requests_lock_.lock();
   // Create local copy of the requests to cancel
-  auto contexts = std::move(task_contexts_);
   auto requests_map = requests_map_;
+  auto contexts = std::move(task_contexts_);
 
   requests_lock_.unlock();
+
   for (auto& pair : requests_map) {
     pair.second.cancel();
   }
 
+  for (auto context : contexts) {
+    context.CancelToken().cancel();
+  }
+
+  return true;
+}
+
+bool PendingRequests::CancelAllAndWait() {
```

Comment:
Functionally, `CancelAllAndWait` is the same as previous `CancelPendingRequests`. To prevent deadlocking we still need to `cancel` and then `BlockingCancel`